src/views/virtualmachines/utils/virtualMachineRowFilter.ts

```diff
@@ -211,10 +211,38 @@ const useIPSearchFilter = (vmiMapper: VMIMapper): RowFilter => ({
   type: 'ip',
 });
 
-const useStorageclassFilter = (vms: V1VirtualMachine[]): RowFilter => {
-  const storageClasses = useMemo(
-    () => Array.from(new Set(vms?.map((vm) => getVirtualMachineStorageClasses(vm)).flat())),
-    [vms],
+type StorageClassByVM = { [namespace in string]: { [name in string]: Set<string> } };
+
+const useStorageclassFilter = (vms: V1VirtualMachine[], pvcMapper: PVCMapper): RowFilter => {
+  const { allStorageClasses, storageClassesByVM } = vms.reduce(
```

Comment:
ideally we should be able to re-use the next version of `getVirtualMachineStorageClasses()` helper for the first step  i.e.
```
vms
  .map( vm => [name,ns,getVirtualMachineStorageClasses(vm, pvcMapper)]
  .reduce( ...)